src/integrationTest/java/org/opensearch/security/privileges/dlsfls/FlsFmIntegrationTests.java

```diff
@@ -0,0 +1,890 @@
+            // ~[lucene-core-10.1.0.jar:10.1.0 884954006de769dc43b811267230d625886e6515 - 2024-12-17 16:15:44]
+            // at org.apache.lucene.search.TopFieldCollector$SimpleFieldCollector$1.collect(TopFieldCollector.java:200)
+            // ~[lucene-core-10.1.0.jar:10.1.0 884954006de769dc43b811267230d625886e6515 - 2024-12-17 16:15:44]
+            // at org.apache.lucene.search.Weight$DefaultBulkScorer.scoreRange(Weight.java:324) ~[lucene-core-10.1.0.jar:10.1.0
+            // 884954006de769dc43b811267230d625886e6515 - 2024-12-17 16:15:44]
+            // at org.apache.lucene.search.Weight$DefaultBulkScorer.score(Weight.java:264) ~[lucene-core-10.1.0.jar:10.1.0
+            // 884954006de769dc43b811267230d625886e6515 - 2024-12-17 16:15:44]
+            // at org.opensearch.search.internal.CancellableBulkScorer.score(CancellableBulkScorer.java:71)
+            // ~[opensearch-3.0.0-beta1-SNAPSHOT.jar:3.0.0-beta1-SNAPSHOT]
+            // at org.opensearch.search.internal.ContextIndexSearcher.searchLeaf(ContextIndexSearcher.java:356)
+            // ~[opensearch-3.0.0-beta1-SNAPSHOT.jar:3.0.0-beta1-SNAPSHOT]
+            // at org.opensearch.search.internal.ContextIndexSearcher.search(ContextIndexSearcher.java:305)
+            // ~[opensearch-3.0.0-beta1-SNAPSHOT.jar:3.0.0-beta1-SNAPSHOT]
+            // at org.opensearch.search.internal.ContextIndexSearcher.search(ContextIndexSearcher.java:269)
+            // ~[opensearch-3.0.0-beta1-SNAPSHOT.jar:3.0.0-beta1-SNAPSHOT]
+            // at org.opensearch.search.query.QueryPhase.searchWithCollector(QueryPhase.java:355)
+            // ~[opensearch-3.0.0-beta1-SNAPSHOT.jar:3.0.0-beta1-SNAPSHOT]
+            return;
+        }
+
+        try (TestRestClient client = cluster.getRestClient(user)) {
+            TestRestClient.HttpResponse response = client.postJson(TEST_INDEX.name() + "/_search", """
+                {
+                    "sort" : [
+                       { "attr_keyword": "asc"},
+                       { "attr_long" : "desc" }
+                    ]
+                }""");
+
+            // TODO: Assert for case where attr_keyword is hidden by FLS an assertion that no values are exposed in the sort attribute of
+            // hits
+            assertThat(response, isOk());
+            assertThat(
+                response,
+                hasSearchHits(whereDocumentSourceEquals(TEST_DOCUMENTS.applyTransform(user.reference(DOC_WITH_FLS_FM_APPLIED))))
+            );
+        }
+    }
+
+    @Test
+    public void termVectors() {
+        if (user == Users.MASKING_ON_TEXT
+            || user == Users.MASKING_ON_KEYWORD
+            || user == Users.MASKING_ON_BINARY
+            || user == Users.MASKING_ON_IP
+            || user == Users.MASKING_ON_STORED_FIELD) {
+            // Term vectors on field masking protected documents fail with an internal OpenSearch assertion error
```

Comment:
Should this test be updated now that https://github.com/opensearch-project/security/pull/5243 is merged?

Reply:
reported issue